src/shogun/machine/Machine.h

```diff
@@ -25,6 +25,20 @@
 namespace shogun
 {
 
+#define TRAIN_DENSE_DISPATHER(train_templated_name) \
+virtual bool train_with_dense(CFeatures* data) \
+{  \
```

Comment:
As of now, I am quite puzzled how we could combine this idea with iterative machine, since making the `iteration` method virtual *and* templated is not an option, but neither is to do the dispatching inside that method :(

Reply:
Could we make the macro in a way that we can use it do overload the 'train_with_dense' to dispatch the template type, but also that we can use it to overload 'iterate_dense' ?

Reply:
i think we will need a new macro for iterate_dense

Reply:
the code is essentially the same, so why not try to create one macro for both.

* accept the to be overloaded name as a parameter, here `train_with_dense`
* accept a variable number of parameters that are passed on to the definition of the overloaded method and the call of the typed implementation

Reply:
nope, but it would be cool to use the same macro